https://issues.redhat.com/browse/ACM-28722-topic #1-3: Update migration doc to su… (#8657)
open-cluster-management/rhacm-docs: global_hub/global_hub_ztp_migration.adoc

[#ztp-migration-overview]
= Zero Touch Provisioning migration overview

Zero Touch Provisioning (ZTP) migration facilitates the transfer of fully provisioned, bare-metal clusters, such as {sno}, from a source {acm-short} hub cluster to a target hub cluster. Migration maintains the cluster state, configuration, and GitOps management throughout the transition.

> //is this GitOps in general or OCP GitOps?

Author: see my comment here line 5

Reviewer: OCP GitOps

| Referenced secrets
| Credentials and pull secrets that the system collects automatically.

| BMC credentials
| Secrets referenced in the `BareMetalHost.spec.bmc.credentialsName` field.

| Pull secrets
| Secrets that are referenced in the `ClusterDeployment.spec.pullSecretRef.name` field.
|===

The following config maps are included in the migration:

Referenced config maps:: Automatically collects config maps that are referenced.
Extra Manifests:: Config maps that are referenced in `ImageClusterInstall.spec.extraManifestsRefs`

The following bare metal resources with status preserved are included in the migration:

BareMetalHost:: Physical server inventory and state
HostFirmwareSettings:: BIOS and firmware configurations
FirmwareSchema:: Firmware setting schemas
HostFirmwareComponents:: Individual firmware components

The following resources are excluded from migration:

ClusterInstance Resources:: Managed by Argo CD, you must manually apply GitOps applications to the target hub cluster after cluster migration.
Policy Resources:: Must be redeployed by GitOps applications on target hub cluster.
Application Resources:: GitOps applications managing cluster configuration.

> //I don't udqerstand this last one. 

Author: comment on line 100

Reviewer: The application means the Customer Resource which used to sync policy/clusterinstance/other cluster configuraiton, it will be managed by Openshfit GitOps Operator.
For example:
https://github.com/openshift-kni/telco-reference/blob/release-4.20/telco-ran/configuration/argocd/deployment/clusters-app.yaml#L9

[#ztp-migration-overview]
= Zero Touch Provisioning migration overview

Zero Touch Provisioning (ZTP) migration facilitates the transfer of fully provisioned, bare-metal clusters, such as {sno}, from a _source_ {acm-short} hub cluster to a _target_ hub cluster. Migration maintains the cluster state, configuration, and {gitops-short} management throughout the transition. 

== Resources that are included in migration

The ZTP migration procedure automatically transfers several resource categories from the source hub cluster to the target hub cluster to ensure continuity with the following processes:

> Pre-migration verification:: Verify the clusters on the source hub cluster to ensure they are ready for migration.

Reviewer 2: recommending to remove "they"
```suggestion
Pre-migration verification:: Verify the clusters on the source hub cluster to ensure migration readiness.
````

Author: good catch @dockerymick

Create migration resource:: Manually create the migration resource on the global hub cluster to initiate the process.
Migration execution:: The {global-hub} cluster automatically transfers resources and registers clusters to the target hub cluster.
Apply `ClusterInstance` applications:: Manually apply the `ClusterInstance` applications to the target hub cluster.
Verify migration:: On the target hub cluster, verify that all clusters are available, test cluster operations, and confirm policy compliance.
Cleanup source hub cluster:: Manually delete the {gitops-short} applications and the `ClusterInstance` resources from the source hub cluster.

== Supported versions of {acm-short} and {ocp-short} for migration

See the following version compatibility for migration:

.Version requirements for hub cluster components
|===
| Component | Source hub cluster | Target hub cluster| Notes

| {acm-short}
| Version `N`
| Version `N to N+1`, and EUS version support
| None

| {ocp-short}
| Version `4.x`
| Version `4.x`
| Compatible with {acm-short} version

| {global-hub}
| Latest stable

> | None

Reviewer 3: Swap "Latest stable" and "None", "Latest stable" put "Target hub cluster" column.

Author: It may be better to commit what you need here using the Add a suggestion icon if this is not the change.

| Installed on the target hub cluster only
|===

== Core resources that are included

The following resources are included in the migration process:

`ManagedCluster`:: Facilitates cluster registration and management.
`KlusterletAddonConfig`:: Defines add-on configurations for managed clusters.

== Deployment resources that are included

The following deployment resources are included in the migration:

`ClusterDeployment`::
Contains Hive deployment configuration while preserving status.
`ImageClusterInstall`::
Manages image-based installation configuration while preserving status.

== Secrets that are included

The following secrets are included in the migration:

.Secrets that are included in the migration
|===
| Secret type | Description

| Administrative credentials
| The `<cluster-name>-admin-password` secret.

| Administrative Kubeconfig
| The `<cluster-name>-admin-kubeconfig` secret.

| Metadata
| The `<cluster-name>-metadata-json` secret.

| Seed reconfiguration
| The `<cluster-name>-seed-reconfiguration` secret.

| Network secrets
| Secrets that contain the `siteconfig.open-cluster-management.io/sync-wave` annotation.

> | Referenced secrets

Reviewer: Removed the `Referenced secrets`

Author: done

| Credentials and pull secrets that the system collects automatically.

| BMC credentials
| Secrets referenced in the `BareMetalHost.spec.bmc.credentialsName` field.

| Pull secrets
| Secrets that are referenced in the `ClusterDeployment.spec.pullSecretRef.name` field.
|===

== Config maps that are included 

The following config maps are included in the migration:

> Referenced config maps:: Automatically collects config maps that are referenced.

Reviewer: Removed the Referenced config maps

Author: done

Extra Manifests:: Config maps that are referenced in `ImageClusterInstall.spec.extraManifestsRefs`

== Bare metal resources that are included 

The following bare metal resources with status preserved are included in the migration:

`BareMetalHost`:: Physical server inventory and state
`HostFirmwareSettings`:: BIOS and firmware configurations
`FirmwareSchema`:: Firmware setting schemas
`HostFirmwareComponents`:: Individual firmware components

== Resources that are _not_ included in the migration

`ClusterInstance` Resources:: Resources that are managed by Argo CD. Manually apply {gitops-short} applications to the target hub cluster after cluster migration.
`Policy Resources`:: Resources that are used for polices. Use {gitops-short} applications to re-deploy on the target hub cluster.

> `Application Resources`:: Configuration for {gitops-short} applications managing clusters. 

Author: For each of these except the last one, we have an action:

_Manually apply {gitops-short} applications to the target hub cluster after cluster migration._ (for example)

To be consistent and for UX, I would like to do the same for:

`Application Resources`:: Configuration for {gitops-short} applications managing clusters. 

@ldpliu can you help?
